src/v3/src/transformer/field/transform.ts

```diff
@@ -92,7 +92,11 @@ export const transformStepInputs = (
       if (required && mutable !== false && type !== 'object') {
         acc.dataSchema[name] = {
           validate(data) {
-            const isValid = !!data[name];
+            // Phone number fields contain +[countryCode] characters by default
```

Comment:
I'm having a hard time understanding this comment easily, can you clarify?

Reply:
W.r.t Phone Numbers, we cannot just blindly assert that the field is set, the reason for this is because phone numbers will always have a `+` and country code, set as the default value no matter what (this is set in PhoneAuthenticator.tsx). If we just assert on the fact that it is empty or not for validation, it will never block the user from leaving an empty field and will allow them to submit without entering a phone number in the phone number input field. 

Want me to update the comment in the code?

Reply:
Ah, okay, that's what I suspected but I wanted to be sure. I think a comment that's more like "contain the '+' and country code characters by default" is clearer.

As a followup, we assert only `> 2` but if the country code is longer, won't this fail to properly prevent submitting an empty input field? e.g. U.K. is `+44` which is 3 characters.

Reply:
Oh yea, I think I need some regex here to cover all country code scenarios 😮‍💨 Will update that.

Reply:
wait, iirc, we discussed to not do client side validation for phone number for now, and only follow regex pattern from backend when it's available in the response.

Reply:
and changes should not happen in the generic field transformation step

Reply:
This is only for `empty` field validation. You can't just blindly check the existence of data in the phone number field because by default the phone number will always have `+1` or `+[whatever country code is selected]` so it will always pass the validation check.

Reply:
then reassign dataschema for phone transformer, not here

Reply:
Regardless of what flow the phone number shows up in, this validation check rule should always apply, so doesn't that mean it belongs here? This logic is for setting up the validation rule for 'required' fields, when phone number is a required field, it just had different logic to validate whether it is set or not. Just curious?

Reply:
generic logic should not know anything about phone number, if there is a special scenario, handle it in that specific transformer, or dataSchema step.

Reply:
👍 Ok, I will move to dataSchema.

Reply:
with handling it in specific transformer, I think you can just get the country code and remove it from data, so no regex should be needed. bad regex blocks user from finishing the flow

Reply:
Even in custom transformers, we wont know what part of the phone number is the country code as it is all just appended as one single string. 

`+2162` vs `+1216` vs `+202`

How will the transformer know what part of this string is a country code?

I could use the countryCodeUtil to perform a match on the string to extract the country code, and then assert on the remaining part of the string to determine if a value is set or not. Is that what you meant?

Reply:
possible to update dataSchema for phonenumber field to include countryCode while it changes, then pass it as options to validate function. then you should have enough info to proceed with required validation.

Reply:
maybe it's not a bad idea to separate this feature into another PR, and keep this pr focused

Reply:
Ok, this is going to require some complex logic, I will remove this change and create a separate PR specifically for phone number validation.